grpclb/src/main/java/io/grpc/grpclb/GrpclbNameResolver.java

```diff
@@ -110,9 +110,10 @@ protected InternalResolutionResult doResolve(boolean forceTxt) {
         }
       }
     }
-    if (srvRecordsException != null
-        || (balancerAddressesException != null && balancerAddresses.isEmpty())) {
-      logger.log(Level.FINE, "Balancer resolution failure", srvRecordsException);
+    if (srvRecordsException != null) {
+      logger.log(Level.FINE, "SRV lookup failure", srvRecordsException);
+    } else if (balancerAddressesException != null && balancerAddresses.isEmpty()) {
+      logger.log(Level.FINE, "Balancer resolution failure", balancerAddressesException);
```

Comment:
Note that this still would cause the same confusion as you had. I think you want to change this string as well, maybe to something like "SRV-provided hostname lookup failure"

Reply:
Yeah, done.